src/Components/Publishing/Byline/Share.tsx

```diff
@@ -1,20 +1,27 @@
 import React from "react"
 import styled from "styled-components"
-import { track } from "../../../Utils/track"
+import Events from "../../../Utils/Events"
+import track from "../../../Utils/track"
 import { pMedia } from "../../Helpers"
 import { IconSocialEmail } from "../Icon/IconSocialEmail"
 import { IconSocialFacebook } from "../Icon/IconSocialFacebook"
 import { IconSocialTwitter } from "../Icon/IconSocialTwitter"
 
-interface ShareProps extends React.HTMLProps<HTMLDivElement> {
+interface Props extends React.HTMLProps<HTMLDivElement> {
   url: string
   title: string
   articleId?: string
   color?: string
+  tracking?: any
+  trackingData?: any
 }
 
-@track()
-export class Share extends React.Component<ShareProps, null> {
+@track((props) => {
+  return props.trackingData ? props.trackingData : {}
+}, {
+  dispatch: data => Events.postEvent(data)
+})
+export class Share extends React.Component<Props, null> {
   static defaultProps = {
     color: "black",
   }
```

Comment:
Does this work without setting tracking in defaultProps?

Reply:
I believe so, this is what I see in storybooks - 
![screen shot 2017-11-10 at 5 52 37 pm](https://user-images.githubusercontent.com/1497424/32681940-021319fa-c640-11e7-9fed-517013a34a05.png)

Reply:
Added a story for the custom props as well, looks to be working - - 
<img width="777" alt="screen shot 2017-11-12 at 1 37 46 pm" src="https://user-images.githubusercontent.com/1497424/32702072-f09ba5be-c7ae-11e7-8eb0-25409d21d2b7.png">